source/Ship.cpp

```diff
@@ -173,7 +178,60 @@ void Ship::Load(const DataNode &node)
 				hasEngine = true;
 			}
 			enginePoints.emplace_back(.5 * child.Value(1), .5 * child.Value(2),
-				(child.Size() > 3 ? child.Value(3) : 1.));
+				(child.Size() > 3 ? child.Value(3) : 1.), (child.Size() > 4 ? child.Value(4) : 0.));
```

Comment:
It might be easier to read these constructor calls if we compute the angle first.
Having more than one conditional operator per expression can get unwieldy fast

Reply:
As per the comments in #4749, this syntax is probably changing anyways.

Reply:
It makes some sense to create an intermediate class that is a bit more involved than just `Point`, that the Bay and EnginePoint derive from or contain. This class would handle the angle, facing, and z height child nodes, both read and write. (For compatibility Bay and EnginePoint will still need to parse their opening line.)
A complication here is the "launch effect" child node for Bays, since that can't be parsed by the superclass. It's also reasonable to expect that additional child nodes are added to either Bay or EP that are exclusive to their context. (Perhaps only "retrieve effect", but who knows.) In this sense, it might make sense to make the angle, facing, and z nodes be a child of their own tag, (e.g.`point`), which also serializes / deserializes the x, y, and zoom. This would make  engine specifications a little strange, since that's all they would be:
```
"[steering |reverse ]engine" [optional legacy keywords]
  point x y
    angle A
    drawn under|over
    zoom Z
```

Are there other features that could make sense on an engine specification that aren't discussed here? If so, that would be justification for being a little "wordy" like the above.

Reply:
> It makes some sense to create an intermediate class that is a bit more involved than just `Point`, that the Bay and EnginePoint derive from or contain.

Should that be done in this PR, in #4749, or merge both and then make this change?

Reply:
My comment above is something that should be addressed before merging either this or the other PR. I'm interested to hear thoughts from you, @petervdmeer, and other content creators

Reply:
I'm leaning more away from this idea than toward it. The only things that engines and bays share between this PR and Peter's are the over/under and the angle. They also share their x/y coordinates, but that's also shared with guns and turrets, so I don't see why we'd want to make the syntax for defining bays and engines different from guns and turrets. Outside of that, only bays have their launch effect, and only engines have zoom (while only steering has its steering direction). So really we'd only be making this extra class to handle over/under and angles, which seems unnecessary to me given that we're only dealing with bays and engines.

Reply:
Having the same syntax on engines, bays, guns and turrets will help content creators, so I'm happy that Amazinite took some action to get the syntaxes aligned.

Setting angles, under/over and zoom directly on the engine or bay feels more natural for content creation than adding those under a point, so changing the syntax to add the keywords under point is a NO for me.

Creating a common class is a good idea, but it also looks a bit too early for me; we are still in the process of getting to a more common syntax and let's first see how far we get with that (#4801 will also add angles to guns, making the gun also a little more common in syntax).